ergo-wallet/src/main/scala/org/ergoplatform/wallet/interpreter/ErgoInterpreter.scala

```diff
@@ -82,12 +87,62 @@ class ErgoInterpreter(params: ErgoLikeParameters)(implicit IR: IRContext)
+    /*15540*/"100c0e20dcd3cdd11102e7cb675fd5185a1685ad007619badcf398864274b137d9f45a9e0e20f8a56076694fe7252fc49c8c7c2d99b878d6146a2f8a1b607793bfa1407a32380402040005c08db701040404000404040205c08db70104060400d805d601d901010493cbc2b2a47201007300d602c5a7d603c2a7d604e4c6a70407d6057301eb02d1ecedededda7201017302dad901060493c5b2a47206007202017303dad9010604d801d608b2a5720600eded93c27208720393c1720899c1a7730493e4c6720804077204017305dad901066393cbc27206720501b2a5730600edededda7201017307dad901060493c5b2a47206007202017308dad9010604d801d608b2a5720600eded93c27208720393c1720899c1a7730993e4c672080407720401730adad901066393cbc27206720501b2a4730b00cd7204",
+    /*14982*/"100c0e20dcd3cdd11102e7cb675fd5185a1685ad007619badcf398864274b137d9f45a9e0e200aa0f3bcb01af09ba6ad3cbe442ca253ebd601fbe915450c7a2165ed179a18b00402040005c08db701040404000404040205c08db70104060400d805d601d901010493cbc2b2a47201007300d602c5a7d603c2a7d604e4c6a70407d6057301eb02d1ecedededda7201017302dad901060493c5b2a47206007202017303dad9010604d801d608b2a5720600eded93c27208720393c1720899c1a7730493e4c6720804077204017305dad901066393cbc27206720501b2a5730600edededda7201017307dad901060493c5b2a47206007202017308dad9010604d801d608b2a5720600eded93c27208720393c1720899c1a7730993e4c672080407720401730adad901066393cbc27206720501b2a4730b00cd7204",
+    /*14435*/"100c0e20dcd3cdd11102e7cb675fd5185a1685ad007619badcf398864274b137d9f45a9e0e2019323d02836f596329b28b2d380d3cf27a10f55db7b1d33bc4537ab538594bae0402040005c08db701040404000404040205c08db70104060400d805d601d901010493cbc2b2a47201007300d602c5a7d603c2a7d604e4c6a70407d6057301eb02d1ecedededda7201017302dad901060493c5b2a47206007202017303dad9010604d801d608b2a5720600eded93c27208720393c1720899c1a7730493e4c6720804077204017305dad901066393cbc27206720501b2a5730600edededda7201017307dad901060493c5b2a47206007202017308dad9010604d801d608b2a5720600eded93c27208720393c1720899c1a7730993e4c672080407720401730adad901066393cbc27206720501b2a4730b00cd7204"
+  )
+
+  /** Script processor which uses [[RuntimeIRContext]] to process graphs.
+    * Preforms pre-compilation of the given scripts during instantiation.
+    * Keeps pre-compiled data structures for the lifetime of JVM.
+    */
+  val scriptProcessor = {
+    val scriptKeys = frequentlyUsedScripts.flatMap { s =>
+      val bytes = ErgoAlgos.decodeUnsafe(s)
+      validationSettings.map(vs => CacheKey(bytes, vs))
+    }
+    new PrecompiledScriptProcessor(
+      ScriptProcessorSettings(
+        predefScripts = scriptKeys,
+        maxCacheSize = 2000,  // takes up around 600Mb
```

Comment:
We have nodes running on Raspberry Pi with 512 MB limit, 600 Mb cache is too big

Reply:
reduced to 500